temporian/implementation/numpy/operators/window/simple_moving_average.py

```diff
@@ -0,0 +1,60 @@
+"""Implementation for the SimpleMovingAverage operator."""
+
+from typing import Dict, Optional
+
+import numpy as np
+
+from temporian.implementation.numpy.data.event import NumpyEvent, NumpyFeature
+from temporian.core.data.duration import Duration
+from temporian.core.operators.simple_moving_average import SimpleMovingAverage
+
+
+class SimpleMovingAverageNumpyImplementation:
+    """Numpy implementation for the simple moving average operator."""
+
+    def __init__(self, op: SimpleMovingAverage) -> None:
+        assert isinstance(op, SimpleMovingAverage)
+        self._op = op
+
+    def __call__(
+        self,
+        event: NumpyEvent,
+        sampling: Optional[NumpyEvent] = None,
+    ) -> Dict[str, NumpyEvent]:
+        # TODO: Add support for index.
+        if sampling is not None:
+            raise NotImplementedError("Sampling not implemented")
+
+        dst_event = NumpyEvent(data={}, sampling=event.sampling)
+
+        # For each index
+        # Note: mts => multi time series
+        # Note: ts => time series
+        for index, src_mts in event.data.items():
+            dst_mts = []
+            dst_event.data[index] = dst_mts
+            src_time = event.sampling.data[index]
+
+            # For each feature
+            for src_ts in src_mts:

```

Comment:
We could also concatenate `src_mts` generating an array of shape `m x n`, where `m` is the number of time series and `n` the number of samples of each series. We could then vectorize the entire implementation instead of looping over each time series independently. 

I'm not exactly sure about the overhead concatenating the time series introduces vs. the performance we gain from the vectorization (and how it scales with the number and length of the series) though. We can research about it.

Reply:
Some thoughts:
- Different time series might have different number of observations.
- When using np, it is easy to write memory inefficient algorithms. For example, the current algorithm memory is O(n^2) with n the maximum time series length. This is too large (ultimately, we don't want to exceed O(mn); with m the number of observations in a window). I think the memory complexity of your proposed algorithm is O(n^2p) with p the number of time series.